### ISSUE TITLE
Simplesamlphp 2.1.0 Error 

### ISSUE THREAD

**Issue by manjuaxxonet** [simplesamlphp/simplesamlphp#1890]
### Summary
_Simplesamlphp vesrion 2.1.0 showing 500 error._

### Details
We have downloaded simplesamlphp 2.1.0 from the official website [https://simplesamlphp.org/download/](https://simplesamlphp.org/download/) and followed the instructions as given in the documentation are as follows: 

cp config/config.php.dist config/config.php
cp config/authsources.php.dist config/authsources.php
cp metadata/saml20-idp-hosted.php.dist metadata/saml20-idp-hosted.php
cp metadata/saml20-idp-remote.php.dist metadata/saml20-idp-remote.php
cp metadata/saml20-sp-remote.php.dist metadata/saml20-sp-remote.php

After that when we try to visit "www.mydomain.com/simplesaml/", we are getting 500 error.

Installed versions of software in my server:
*) Ubuntu 20.04.6 LTS
*) Apache 2.4.57
*) PHP 8.0

### Impact
_500 error_

### Screenshot

![simplesamlphp-error](https://user-images.githubusercontent.com/149997311/280617095-5a11a28f-89e6-4b05-8270-8e4548be5715.png)

**Comment by tvdijen** [comment]
And your Apache-logs are telling you what? There should be an error there related to the 500 error.

**Comment by manjuaxxonet** [comment]
> And your Apache-logs are telling you what? There should be an error there related to the 500 error.

We did not get any error logs in apache (log/apache2/error.log)

**Comment by tvdijen** [comment]
That sounds like misconfiguration to me.. How about your system logs (i.e. /var/log/messages or similar)?

**Comment by manjuaxxonet** [comment]
We are using "php-fpm". There are no log messages related to simplesamlphp.
![image](https://github.com/simplesamlphp/simplesamlphp/assets/149997311/e333dcba-7f17-4c67-bf98-ecc44d3b9026)

**Comment by tvdijen** [comment]
Try grepping your php-fpm configuration for `slowlog` and check the logfile configured there.

**Comment by eddiekelley** [comment]
TL;DR - you might need to install the `php-intl` extension. Although SimpleSAMLphp 2.0.x seems to work without that (required) extension, it causes an exception in 2.1.x.

We were seeing similar issues when using SimpleSAMLphp within the `php:8.2-apache` Docker container (`mod_php`).

When visiting the `/simplesaml/admin` endpoint, the error shown in the original post occurs. Although there is an entry in the `/var/log/apache2/access_log` indicating the 500 error, there are no PHP errors logged to `/var/log/apache2/error_log`.

After configuring SimpleSAMLphp to log to file and after increasing the SimpleSAMLphp's logging configuration to debug, we are seeing the following error in the `simplesamlerror.log`:

```
Nov 13 05:35:40 simplesamlphp DEBUG [e8a374aeb8] Trying langpath for 'en' as '/var/simplesamlphp/modules/admin/locales/en/LC_MESSAGES/'
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] Backtrace:
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 2 /var/simplesamlphp/src/SimpleSAML/Error/ExceptionHandler.php:35 (SimpleSAML\Error\ExceptionHandler::customExceptionHandler)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 1 /var/simplesamlphp/vendor/symfony/error-handler/ErrorHandler.php:541 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 0 [builtin] (N/A)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] Caused by: Symfony\Component\ErrorHandler\Error\ClassNotFoundError: Attempted to load class "IntlDateFormatter" from the global namespace.
Did you forget a "use" statement?
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] Backtrace:
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 7 /var/simplesamlphp/vendor/twig/intl-extra/IntlExtension.php:59 (SimpleSAML\XHTML\Template::setupTwig)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 6 /var/simplesamlphp/src/SimpleSAML/XHTML/Template.php:170 (SimpleSAML\XHTML\Template::__construct)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 5 /var/simplesamlphp/modules/admin/src/Controller/Config.php:124 (SimpleSAML\Module\admin\Controller\Config::main)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 4 /var/simplesamlphp/vendor/symfony/http-kernel/HttpKernel.php:163 (Symfony\Component\HttpKernel\HttpKernel::handleRaw)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 3 /var/simplesamlphp/vendor/symfony/http-kernel/HttpKernel.php:75 (Symfony\Component\HttpKernel\HttpKernel::handle)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 2 /var/simplesamlphp/vendor/symfony/http-kernel/Kernel.php:202 (Symfony\Component\HttpKernel\Kernel::handle)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 1 /var/simplesamlphp/src/SimpleSAML/Module.php:234 (SimpleSAML\Module::process)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] 0 /var/simplesamlphp/public/module.php:17 (N/A)
Nov 13 05:35:40 simplesamlphp ERROR [e8a374aeb8] Error report with id 69c08bc0 generated.
```

[This Google Groups post](https://groups.google.com/g/simplesamlphp/c/Gb05GM5fUKk) is reporting the same error, and suggests installing the `php-intl` extension. Installing that PHP extension resolved the issues for us.

It may be worth noting that SimpleSAMLphp 2.0.x _indicated that the `php-intl` extension was installed, even though it wasn't_.

**Comment by tvdijen** [comment]
I've been working on this project for almost a decade now and the intl-extension has always been a requirement.. It's also listed specifically in the installation documentation. This can only happen when you install by deploying the tarball, because a `composer install` will fail hard if the extension is missing.

**Comment by eddiekelley** [comment]
The [SimpleSAMLphp Documentation](https://simplesamlphp.org/docs/stable/simplesamlphp-install.html) suggests installing and upgrading via tarball, so that's the way that we've been doing it.

I guess I'm just surprised that the software doesn't fail more gracefully (and dump a stack trace like it does with other errors) when the `php-intl` extension is not present.

This made the upgrade process to v.2.1.0 a little more difficult, and was especially perplexing since it didn't log any indication of what the issue was until we enabled debug logging.

We've been using the software for years without that extension, and since reverting to the old version gave the A-OK for that extension in the checklist (even though it wasn't installed) it made this issue harder to troubleshoot.

**Comment by tvdijen** [comment]
> We've been using the software for years without that extension, and since reverting to the old version gave the A-OK for that extension in the checklist (even though it wasn't installed) it made this issue harder to troubleshoot.

I'm not sure how this can happen. Maybe Symfony has changed their behaviour.
I see two options here:
- We use a polyfill for the intl-extension
- We start catching the ClassNotFoundException raised by Symfony and convert it to a more graceful SSP CriticalConfigurationException

@thijskh What are your thoughts on this one?
